src/hit/api/evaluator/homomorphic.cpp

```diff
@@ -236,6 +256,20 @@ namespace hit {
         return backend_encryptor->object;
     }
 
+    Bootstrapper &HomomorphicEval::get_bootstrapper() {
+        if (!(context->btp_params.has_value())) {
+            LOG_AND_THROW_STREAM("CKKS parameters do not specify bootstrapping parameters.");
+        }
+
+        if (backend_bootstrapper.get() == nullptr || backend_bootstrapper->params != context->params) {
+            ParameterizedLattigoType<Bootstrapper> *tmp =
+                new ParameterizedLattigoType<Bootstrapper>(
+                    newBootstrapper(context->params, context->btp_params.value(), btp_keys), context->params);
+            backend_bootstrapper.reset(tmp);
+        }
+        return backend_bootstrapper->object;
+    }
```

Comment:
Need lock?

Reply:
Not needed here. `backend_bootstrapper` is a `thread_specific_ptr`, which means I don't have to worry about two threads using the same `backend_bootstrapper`. This code just ensures that a `backend_bootstrapper` exists for each thread.